modules/services/podman-linux/builds.nix

```
@@ -0,0 +1,173 @@
+      };
+
+      description = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "My Build";
+        description = "The description of the build.";
+      };
+
+      environment = mkOption {
+        type = podman-lib.primitiveAttrs;
+        default = { };
+        example = literalExpression ''
+          {
+            VAR1 = "0:100";
+            VAR2 = true;
+            VAR3 = 5;
+          }
+        '';
+        description = "Environment variables to set in the build.";
+      };
+
+      extraConfig = mkOption {
+        type = podman-lib.extraConfigType;
+        default = { };
+        example = literalExpression ''
+          {
+            Build = {
+              Arch = "aarch64";
+            };
+            Service = {
+              TimeoutStartSec = 15;
+            };
+          }
+        '';
+        description = "INI sections and values to populate the Build Quadlet.";
+      };
+
+      extraPodmanArgs = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "--retries 5" ];
+        description = "Extra arguments to pass to the podman build command.";
+      };
+
+      file = mkOption {
+        type = types.str;
+        example = literalExpression ''
+          `"''${config.xdg.configHome}/containerfiles/my-img/Containerfile"`
+          or
+          `"https://github.com/.../my-img/Containerfile"`
+        '';
+        description =
+          "Path to a Containerfile which contains instructions to build the image.";
+      };
+
+      tags = mkOption {
+        type = with types; listOf str;
+        default = [ ];
+        description = ''
+          Name associated with the build.
+          First tag will always be "homemanager/<name>".
+        '';
+      };
+
+      labels = mkOption {
+        type = with types; attrsOf str;
+        default = { };
+        example = {
+          app = "myapp";
+          some-label = "somelabel";
+        };
+        description = "The labels to apply to the build.";
+      };
+
+      tlsVerify = mkOption {
+        type = types.bool;
+        default = true;
+        description =
+          "Require HTTPS and verification of certificates when contacting registries.";
+      };
+
+      workingDirectory = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = "WorkingDirectory of the systemd unit file.";
+      };
+    };
+  };
+in {
+  options.services.podman.builds = mkOption {
+    type = types.attrsOf buildDefinitionType;
+    default = { };
+    description = "Defines Podman build quadlet configurations.";
+  };
+
+  config = let buildQuadlets = mapAttrsToList toQuadletInternal cfg.builds;
+  in mkIf cfg.enable {
+    services.podman.internal.quadletDefinitions = buildQuadlets;
+    assertions = flatten (map (build: build.assertions) buildQuadlets);
+
+    home.file."${config.xdg.configHome}/podman/images.manifest".text =
```

Not sure why the podman module ended up using this `home.file` format, seems to me that it would be possible to simply use `xdg.configFile`.

```suggestion
    xdg.configFile."podman/images.manifest".text =
```